test/e2e/k8s/operator.go

```diff
@@ -150,6 +152,22 @@ func BuildManager(initCfg *Config) (manager.Manager, error) {
 		return nil, err
 	}
 
+	if err = (&atlasdatafederation.AtlasDataFederationReconciler{
+		ResourceWatcher:             watch.NewResourceWatcher(),
+		Client:                      mgr.GetClient(),
+		Log:                         logger.Named("controllers").Named("AtlasDataFederation").Sugar(),
+		Scheme:                      mgr.GetScheme(),
+		AtlasDomain:                 config.AtlasDomain,
+		GlobalAPISecret:             config.GlobalAPISecret,
+		EventRecorder:               mgr.GetEventRecorderFor("AtlasDataFederation"),
+		GlobalPredicates:            globalPredicates,
+		ObjectDeletionProtection:    config.ObjectDeletionProtection,
+		SubObjectDeletionProtection: config.SubObjectDeletionProtection,
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "AtlasDataFederation")
+		return nil, err
+	}
```

Comment:
I guess this is here because of concurrency with PR #1124 **CLOUDP-195050: Atlas Federated Authentication configuration**

If it requires those changes then it should be rebased after that one.

Reply:
It has nothing to do with Federated Authentication. This adds the Data Federation controller that was missing in the e2e setup. Related tests using Data FEderation CR would never be handled because of this, and actually, the existing Data Federation e2e test has never run.